src/goto-programs/goto_trace.h

```diff
@@ -292,4 +292,28 @@ void show_goto_trace(
   if(cmdline.isset("stack-trace"))                                             \
     options.set_option("stack-trace", true);
 
+/// Variety of constant expression only used in the context of a GOTO trace, to
+/// give both the numeric value and the symbolic value of a pointer.
+class goto_trace_constant_pointer_exprt : public constant_exprt
```

Comment:
I think this clarifies some things. But could you add what you put in the PR description as a documentation here to make it even clearer.

Reply:
https://github.com/diffblue/cbmc/pull/5037